### DISCUSSION TITLE
ICSx5 fails to update subscriptions stating "Unable to resolve host"

### DISCUSSION THREAD

**Discussion by webdb22** [bitfireAT/icsx5#73] [Q&A]
Dear ICSx5 team,

a few subscriptions, which have been working previously, are since a few days (maybe weeks) are often failing with "Unable to resolve host...". This happens only to subscriptions pointing to a web address, local subscriptions work fine.

Just to verify I've tried to put the address used for the subscriptions into a browser on the same device, this always works flawlessly, the address is resolved and the .ics file downloaded. The file size of the downloaded .ics files is rather small, typically in the range of ~10k up to ~100k. Therefore I'm suspecting an issue with the network code or libraries, but that's just a wild guess.

Current environment:
ICSx5 V 2.0.3
Android V 12, Security Patch Level 1. October 2022

Any comment or hint highly welcome, thanks a lot in advance!

**Comment by rfc2822** [comment]
Hi,

I think it's the same as #24 and #72.

Still no idea why this happens...

**Reply by webdb22** [reply]
Hi,

thanks for your prompt reply!

I've reviewed the issues you're mentioning, but as I'm getting an error about resolving the host name, whereas in the others the host name could be resolved but later can't connect to, I've assumed a different issue, but I might be wrong.

Anyway, thanks for looking into it, let me know if I can provide any additional input to help to resolve the issue!

**Reply by rfc2822** [reply]
What is your sync interval? Can you please check

- whether the problem goes away when you "disable battery optimization" for the ICSx5 app in your Android battery settings,
- whether the problem goes away when you set the sync interval to less than 1 day (one time with and one time without "disabling battery optimization")?

**Reply by webdb22** [reply]
For the last week I've watched the behaviour with the current "Unrestricted" battery setting and the sync interval "every hour", the errors happens on average once a day, but some days it works flawlessly, other days it could take hours until a successful sync.  Besides the previous error message "Unable to resolve host..." I'm sometimes seeing "Job was cancelled".

I'm now switching to "optimized" for the Battery Settings and "Once a day" for the sync interval, let's see...

**Reply by webdb22** [reply]
Running with different battery settings and sync intervals for a few weeks now, no change in behaviour, the errors still occur with no noticeable pattern.

**Comment by hrst** [comment]
I'm also having this issue, and it is completely unpredictable as well, the error only occurs sometimes with no apparent pattern or reason.

**Comment by lcasys** [comment]
I'm also having this issue, in my special case it happens about 1 time per month. For me, the update interval is set to 1 time per week.  When the error occurs, I used to get (before Android 13) the message. 'Unable to resolve host…'  Now (with Android 13) I get the message. "Job was cancelled". To fix the error, I have to go to the menu item 'Set update interval', select the chosen interval (1 time per week for me) and save it. If I then run a synchronization. the error is fixed until the next time (as I said, for me about 1 month).

**Comment by gofabian** [comment]
I have the same issue. Most of the time the sync works with 3 calendar subscriptions. A few times a month the sync fails. The error message in the app is for all 3 calendars "Unable to resolve host ...: No address associated with hostname".  The issue can be solved by the "Synchronize now" action.

**Comment by sotel** [comment]
Also here:
**ICSx5** said:

"Unable to resolve host "www.auto-kultur-werkstatt.de": No address associated with hostname"
Please see this screenshot here:

![Screenshot: ICSx⁵](https://github.com/bitfireAT/icsx5/assets/107782/7c8ed7cb-59c9-44c2-b0ed-0adc41bd758f)

Here is **my browser** opening the URL without problems:

![Screenshot: Vivaldi_Browser](https://github.com/bitfireAT/icsx5/assets/107782/806ff66a-c8c0-4436-98b5-845eda6b7d2d)

Works. Also the download of their _ics-file_ direct from their server.

**Comment by sicherist** [comment]
I've had this problem for quite a while. "Synchronize now" doesn't help. What information do you need to debug it? Here is a copy of my error message:

> Unable to resolve host "example.invalid": No address associated with hostname
> 
> java.net.UnknownHostException: Unable to resolve host "example.invalid": No address associated with hostname
> 	at java.net.Inet6AddressImpl.lookupHostByName(Inet6AddressImpl.java:156)
> 	at java.net.Inet6AddressImpl.lookupAllHostAddr(Inet6AddressImpl.java:103)
> 	at java.net.InetAddress.getAllByName(InetAddress.java:1152)
> 	at okhttp3.Dns$Companion$DnsSystem.lookup(Dns.kt:49)
> 	at okhttp3.internal.connection.RouteSelector.resetNextInetSocketAddress(RouteSelector.kt:169)
> 	at okhttp3.internal.connection.RouteSelector.nextProxy(RouteSelector.kt:132)
> 	at okhttp3.internal.connection.RouteSelector.next(RouteSelector.kt:74)
> 	at okhttp3.internal.connection.RealRoutePlanner.planConnect(RealRoutePlanner.kt:147)
> 	at okhttp3.internal.connection.RealRoutePlanner.plan(RealRoutePlanner.kt:67)
> 	at okhttp3.internal.connection.FastFallbackExchangeFinder.launchTcpConnect(FastFallbackExchangeFinder.kt:118)
> 	at okhttp3.internal.connection.FastFallbackExchangeFinder.find(FastFallbackExchangeFinder.kt:62)
> 	at okhttp3.internal.connection.RealCall.initExchange$okhttp(RealCall.kt:267)
> 	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.kt:32)
> 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
> 	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.kt:95)
> 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
> 	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:84)
> 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
> 	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:65)
> 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:109)
> 	at okhttp3.internal.connection.RealCall.getResponseWithInterceptorChain$okhttp(RealCall.kt:205)
> 	at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:537)
> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
> 	at java.lang.Thread.run(Thread.java:1012)
> Caused by: android.system.GaiException: android_getaddrinfo failed: EAI_NODATA (No address associated with hostname)
> 	at libcore.io.Linux.android_getaddrinfo(Native Method)
> 	at libcore.io.ForwardingOs.android_getaddrinfo(ForwardingOs.java:133)
> 	at libcore.io.BlockGuardOs.android_getaddrinfo(BlockGuardOs.java:222)
> 	at libcore.io.ForwardingOs.android_getaddrinfo(ForwardingOs.java:133)
> 	at java.net.Inet6AddressImpl.lookupHostByName(Inet6AddressImpl.java:135)
> 	... 24 more

**Comment by hansdez** [comment]
I also have this issue (running Version 2.1.1-gplay).

For me when I open the app and sync manually (with a pull-down to refresh) the problem never happens. Hope that helps the troubleshooting.

**Reply by returntrip** [reply]
This is not a fix but a possible workaround and an interesting behaviour.

Re adding the calendar link and refreshing will work for the re added calendar. See screenshot:
![Screenshot_20231228_081552_ICSx.jpg](https://github.com/bitfireAT/icsx5/assets/14056122/d36a17e4-535b-4de4-a070-19fc3e2fad08)

**Reply by webdb22** [reply]
I can confirm the behavior stated by @returntrip, if adding the link to a calendar a second time, the "new" entry is resolved fine, whereas the "old" entry - at the same time - shows the error of "Unable to resolve host..."

**Reply by gofabian** [reply]
I had a similar issue. The problem "Unable to resolve host..." appeared from time to time (quite often). But when I started the synchronisation in the app it worked with success.

Some weeks ago I changed the battery settings of the app from "optimised" to "unrestricted". The issue did not appear again.

It seems Android prevented the app from accessing the Internet in the background.

![Screenshot_20240122-214234~2](https://github.com/bitfireAT/icsx5/assets/3092660/0642aaa6-5a30-4c4c-99b9-fcdcb8788754)
